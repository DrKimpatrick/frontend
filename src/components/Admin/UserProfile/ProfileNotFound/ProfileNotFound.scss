.profileNotFound {
  position: relative;
  display: flex;
  flex-direction: column;
  background: white;
  .notFoundContainer {
    width: 380px;
    margin: auto;
    position: relative;
    .details {
      position: relative;
      width: 100%;
      box-shadow: 0px 1px 2px rgba(0, 0, 0, 0.2);
      background: #ebebeb;
      border-radius: 2px;
      padding: 20px;
      padding-bottom: 10px;
      .notFoundImage {
        width: 100%;
        height: 100%;
        img {
          width: 100%;
          height: 100%;
          object-fit: contain;
          border-radius: 5px;
        }
      }
      .message {
        font-size: 20px;
        color: #696969;
      }
    }
  }
}
