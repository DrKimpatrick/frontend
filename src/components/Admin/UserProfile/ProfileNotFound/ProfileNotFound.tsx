import React from 'react';
import './ProfileNotFound.scss';
import { useHistory } from 'react-router-dom';
import Image from 'assets/images/profileNotFound.png';
import { ArrowBack } from '@material-ui/icons';
import { Routes } from 'utils/routes';

export const ProfileNotFound = () => {
  const history = useHistory();

  return (
    <div className="profileNotFound flex justify-center items-center">
      <div className="notFoundContainer">
        <div className="details">
          <div className="notFoundImage">
            <img src={Image} alt="profile not found" />
          </div>
          <div className="message text-center my-5 font-bold">
            Profile is not available
          </div>
        </div>
        <div className="my-5 w-full">
          <button
            type="button"
            onClick={() => history.push(Routes.SuperAdminDashboard)}
            className="flex items-center justify-center text-center w-full"
            style={{ color: '#b3adad', outline: 'none', border: 'none' }}
            id="goBack"
          >
            <ArrowBack />
            <h5 className="mx-3 text-sm">go to dashboard</h5>
          </button>
        </div>
      </div>
    </div>
  );
};

export default ProfileNotFound;
