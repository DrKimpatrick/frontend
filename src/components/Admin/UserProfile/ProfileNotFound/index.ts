import ProfileNotFound from './ProfileNotFound';

export { ProfileNotFound };
