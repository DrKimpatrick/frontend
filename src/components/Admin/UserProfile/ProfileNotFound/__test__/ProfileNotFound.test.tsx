import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import { cleanup, fireEvent, render } from '@testing-library/react';
import ProfileNotFound from '../ProfileNotFound';
import { MemoryRouter } from 'react-router-dom';

const mockHistory = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistory,
  }),
}));

describe('ProfileNotFound', () => {
  afterEach(cleanup);

  it('should render without crashing', () => {
    const div = document.createElement('div');

    ReactDom.render(
      <MemoryRouter>
        <ProfileNotFound />
      </MemoryRouter>,
      div
    );
  });

  it('should create snapshot', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <ProfileNotFound />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should allow user to go back to dashboard', () => {
    const { container } = render(
      <MemoryRouter>
        <ProfileNotFound />
      </MemoryRouter>
    );

    const button: Element | any = container.querySelector('.w-full #goBack');

    fireEvent.click(button);
    expect(mockHistory).toBeDefined();
    expect(mockHistory).toBeCalled();
  });
});
