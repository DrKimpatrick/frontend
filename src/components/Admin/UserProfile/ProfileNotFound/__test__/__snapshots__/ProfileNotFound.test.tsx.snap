// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileNotFound should create snapshot 1`] = `
<div
  className="profileNotFound flex justify-center items-center"
>
  <div
    className="notFoundContainer"
  >
    <div
      className="details"
    >
      <div
        className="notFoundImage"
      >
        <img
          alt="profile not found"
          src="profileNotFound.png"
        />
      </div>
      <div
        className="message text-center my-5 font-bold"
      >
        Profile is not available
      </div>
    </div>
    <div
      className="my-5 w-full"
    >
      <button
        className="flex items-center justify-center text-center w-full"
        id="goBack"
        onClick={[Function]}
        style={
          Object {
            "border": "none",
            "color": "#b3adad",
            "outline": "none",
          }
        }
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
        <h5
          className="mx-3 text-sm"
        >
          go to dashboard
        </h5>
      </button>
    </div>
  </div>
</div>
`;
